java/spark-1.5.2/collectors.scala

```diff
@@ -30,7 +30,11 @@ object Collectors {
   def collectBooleanArr(row: Row, idx: Int): Array[Int] = {
     val arr = row(idx).asInstanceOf[scala.collection.mutable.WrappedArray[_]]
 
-    arr.map(extractBoolean).toArray
+    if (arr == null) {
```

Comment:
TODO: should add unit tests covering each of those cases as existing unit tests don't seem to cover any of those : /

Reply:
Done.